client/web/src/enterprise/batches/settings/BatchSpecExecutionsPage.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useCallback } from 'react'
+import { RouteComponentProps } from 'react-router'
+
+import {
+    FilteredConnection,
+    FilteredConnectionQueryArguments,
+} from '@sourcegraph/web/src/components/FilteredConnection'
+import { Container, PageHeader } from '@sourcegraph/wildcard'
+
+import { PageTitle } from '../../../components/PageTitle'
+import { BatchSpecExecutionsFields } from '../../../graphql-operations'
+
+import { queryBatchSpecExecutions as _queryBatchSpecExecutions } from './backend'
+import { BatchSpecExecutionNode, BatchSpecExecutionNodeProps } from './BatchSpecExecutionNode'
+import styles from './BatchSpecExecutionsPage.module.scss'
+
+export interface BatchSpecExecutionsPageProps extends Pick<RouteComponentProps, 'history' | 'location'> {
+    /** For testing purposes only. */
+    queryBatchSpecExecutions?: typeof _queryBatchSpecExecutions
+    /** For testing purposes only. Sets the current date */
+    now?: () => Date
+}
+
+export const BatchSpecExecutionsPage: React.FunctionComponent<BatchSpecExecutionsPageProps> = ({
+    history,
+    location,
+    queryBatchSpecExecutions = _queryBatchSpecExecutions,
+    now,
+}) => {
+    const query = useCallback(
+        (args: FilteredConnectionQueryArguments) => {
+            const passedArguments = {
+                first: args.first ?? null,
+                after: args.after ?? null,
+            }
+            return queryBatchSpecExecutions(passedArguments)
+        },
+        [queryBatchSpecExecutions]
+    )
+
+    return (
+        <>
+            <PageTitle title="Batch spec executions" />
+            <PageHeader headingElement="h2" path={[{ text: 'Batch spec executions' }]} className="mb-3" />
+            <Container>
+                <FilteredConnection<BatchSpecExecutionsFields, Omit<BatchSpecExecutionNodeProps, 'node'>>
+                    history={history}
+                    location={location}
+                    nodeComponent={BatchSpecExecutionNode}
+                    nodeComponentProps={{ now }}
+                    queryConnection={query}
+                    hideSearch={true}
+                    defaultFirst={20}
+                    noun="execution"
+                    pluralNoun="executions"
+                    listClassName={styles.executionsGrid}
+                    listComponent="div"
+                    className="filtered-connection__centered-summary"
+                    headComponent={ExecutionsHeader}
+                    cursorPaging={true}
+                    noSummaryIfAllNodesVisible={true}
+                    // TODO: This list is just for admins and is not public yet but we
+                    // should think about what to show a new Sourcegraph admin when this
+                    // list is empty.
+                    emptyElement={<>There's nothing here yet!</>}
```

Comment:
I think we could make this a little more focused:

```suggestion
                    emptyElement={<>Nobody has executed a batch change yet!</>}
```